### PR TITLE
Fix type parameter for expectWorker accepting Worker instance.

### PR DESCRIPTION
**Pull request by zach-klippenstein**
The worker type should be unbounded, _not_ the same as the _Workflow's_ output type.

### CODE REVIEW

**Comment by zach-klippenstein** [comment]
Fixed the `EmittedOutput` type.

File: kotlin/workflow-testing/src/main/java/com/squareup/workflow/testing/RenderTester.kt [hunk 1]
```diff
@@ -265,9 +265,9 @@ interface RenderTester<PropsT, StateT, OutputT : Any, RenderingT> {
 /* ktlint-disable parameter-list-wrapping */
 fun <PropsT, StateT, OutputT : Any, RenderingT>
     RenderTester<PropsT, StateT, OutputT, RenderingT>.expectWorker(
-  doesSameWorkAs: Worker<OutputT>,
+  doesSameWorkAs: Worker<*>,
   key: String = "",
-  output: EmittedOutput<OutputT>? = null
+  output: EmittedOutput<Any?>? = null
```
Comment (tehras) [inline kotlin/workflow-testing/src/main/java/com/squareup/workflow/testing/RenderTester.kt]: Why is this required?

Comment (zach-klippenstein) [reply kotlin/workflow-testing/src/main/java/com/squareup/workflow/testing/RenderTester.kt]: This is how you "fake" an output from this worker. This type is the type of the output from the worker.

**Comment by tehras** [comment]
Oh duhhh. Makes sense

File: kotlin/workflow-testing/src/main/java/com/squareup/workflow/testing/RenderTester.kt (hunk 1 above)
Comment (tehras) [reply kotlin/workflow-testing/src/main/java/com/squareup/workflow/testing/RenderTester.kt]: Oh duhhh.
